google/cloud/spanner/internal/session_pool.cc

```diff
@@ -162,29 +155,62 @@ void SessionPool::Release(Session* session) {
   }
 }
 
-StatusOr<std::vector<std::unique_ptr<Session>>> SessionPool::CreateSessions(
-    int num_sessions) {
+// Creates `num_sessions` on `channel` and adds them to the pool.
+//
+// Requires `lk` has locked `mu_` prior to this call. `lk` will be dropped
+// while the RPC is in progress and then reacquired.
```

Comment:
Can we refactor somehow so that the caller drop/retakes the lock?  That is, so that all the "inside the lock" stuff is in `Allocate()`?